classes/emailTemplate/DAO.php

```diff
@@ -281,6 +285,20 @@ public function installEmailTemplates(
                     $this->installAlternateEmailTemplates($contextId, $attrs['key']);
                 }
             }
+
+            if ($recordTemplateGroupAccess) {
+                // Default to true if `isUnrestricted` is not set.
+                $isUnrestricted = $attrs['isUnrestricted'] ?? '1';
```

Comment:
I think this check is enough to cover any plugin-related issues or the email template XML not having this flag. I don't think you need the `$recordTempalteGroupAccess` parameter at all.